### DISCUSSION TITLE
我如何在onerror 捕获到401的http状态后，重新刷新token后，继续执行原有的接口

### DISCUSSION THREAD

**Discussion by heianxiyi** [alovajs/alova#257] [Q&A]
![image](https://github.com/alovajs/alova/assets/22736955/a2fc1957-027b-4e32-944e-ca7bb92c1655)

**Comment by JOU-amjs** [comment]
https://github.com/alovajs/alova/issues/92

**Reply by heianxiyi** [reply]
...
async onSuccess(response, method) {
  // ...
  else if (code === ResultEnum.TOKEN_EXPIRED) {
    await refreshToken();
    const resData= await method.send();
    return resData;
  }
}
...
这个可能不满足我的情况，
<img width="736" alt="image" src="https://github.com/alovajs/alova/assets/22736955/f2cf499a-7873-4fae-8e98-1fd9f6016055">
他会直接走onError，并且
<img width="1484" alt="image" src="https://github.com/alovajs/alova/assets/22736955/f8de6203-b3e6-42fe-b206-c54d6540cb8e">
多个请求同时进行的时候，怎么让后面的等待呢？

**Reply by heianxiyi** [reply]
<img width="919" alt="image" src="https://github.com/alovajs/alova/assets/22736955/295e2303-d4d5-413f-a99d-6514df95f23d">
改好的，如果是使用onSuccess和onError的情况下，如何刷token

**Reply by JOU-amjs** [reply]
现在，我们提供了专门用于token认证的请求模块，你可以直接使用即可，不用再费儿八劲地自己写了。https://alova.js.org/tutorial/strategy/tokenAuthentication
